spotify_tensorflow/flags.py

```diff
@@ -30,20 +31,35 @@ class Flags(object):
     @staticmethod
     def register_dataset_flags():
         logging.info("Registering Dataset flags")
+
+        flags.DEFINE_string("train_subdir", "train",
+                            "Location of training TFRecords, with the training set dir.")
+
+        flags.DEFINE_string("eval_subdir", "eval",
+                            "Location of eval TFRecords, with the training set dir.")
+
+        # Dataset API parameters
+
         flags.DEFINE_integer("batch_size", 128,
-                             "Size of the batch of the dataset iterator.")
+                             "Size of the batch of the dataset iterator. 0 means no batching.")
 
-        flags.DEFINE_integer("buffer_size", 512,
-                             "Size of the buffer of the dataset iterator.")
+        flags.DEFINE_integer("shuffle_buffer_size", 512,
+                             "Size of the shuffle buffer. 0 means shuffle is turned off.")
 
         flags.DEFINE_integer("take_count", -1,
                              "Creates a `Dataset` with at most `count` batches from this dataset.")
 
-        flags.DEFINE_string("train_subdir", "train",
-                            "Location of training TFRecords, with the training set dir.")
+        flags.DEFINE_integer("parsing_threads", mp.cpu_count(),
+                             "Number of threads used for parsing files.")
 
-        flags.DEFINE_string("eval_subdir", "eval",
-                            "Location of eval TFRecords, with the training set dir.")
+        flags.DEFINE_integer("interleaving_threads", 2,
```

Comment:
I wonder if this could be a function of `cpu_count` like `floor(cpu_count/2)`?

Reply:
I haven't played with other machines but I found this parameter to be extremely sensitive, like going from 2 to 3 lower IO throughput on my machine by 200%. 
I think we should be conservative with this one until further consideration. (I was actually quite tempted to turn it off by default)